lib/Backend/IRBuilder.cpp

```diff
@@ -2033,9 +2033,12 @@ IRBuilder::BuildProfiledReg2(Js::OpCode newOpcode, uint32 offset, Js::RegSlot ds
 
     Js::OpCodeUtil::ConvertNonCallOpToNonProfiled(newOpcode);
 
+    Assert(newOpcode == Js::OpCode::BeginSwitch);
```

Comment:
nit: If the function argument is going to be restricted to this extent, it's probably worth it to just remove the argument.